common/remote/rmtsmtp.cpp

```diff
@@ -633,6 +648,74 @@ class CMailInfo
         out.append("HELO ").append(mailServer.get()).append("\r\n");
     }
 
+    void getEhlo(StringBuffer & out) const
+    {
+        out.append("EHLO ").append(mailServer.get()).append("\r\n");
+    }
+
+    void getEhloAndOptions(StringBuffer & out)
+    {
+        // Initial EHLO and gather reply, which should contain ESMTP options
+        getEhlo(out);
+        writeAndAck(out.str(), out.length());
+
+#if defined(_USE_OPENSSL)
+        bool useTLS = false;
+        if (inlen > 0)
+        {
+            // Walk the buffer, looking for the STARTTLS option
```

Comment:
Worth commenting that 250 responses must come first.  Given the comment my expectation was that this would keep skipping to the next line through the buffer.

Reply:
250 codes should be the only non-error responses at this point, but the options that are returned can be in any order. So really, this is a line-by-line scan, aborting as soon as we can process '250 STARTTLS' or '250-STARTTLS'.